modules/usage/module.php

```diff
@@ -484,9 +538,58 @@ private function remove_from_global( $document ) {
 			}
 		}
 
-		update_option( self::OPTION_NAME, $global_usage, false );
+		update_option( self::ELEMENTS_OPTION_NAME, $global_usage, false );
+
+		$document->delete_meta( self::ELEMENTS_META_KEY );
+	}
+
+	/**
+	 * Get document settings usage.
+	 *
+	 * @param Document $document
+	 * @return array
+	 */
+	private function get_document_settings_usage( $document ) {
+		$usage  = [];
+
+		// Ensure not from cache.
+		$document = Plugin::$instance->documents->get( $document->get_id(), false );
+		$controls = $document->get_controls();
+
+		foreach ( $document->get_settings() as $setting_name => $setting_value ) {
+			if ( isset( $controls[ $setting_name ] ) ) {
+				$control = $controls[ $setting_name ];
+				$control_default_value = $controls[ $setting_name ]['default'];
+
+				$is_default = $control_default_value === $setting_value;
+				$is_repeater = is_array( $setting_value ) && isset( $control['fields'] );
+
+				// Some repeater settings have empty controls, which are not exist in the default, it should be analyzed.
+				if ( $is_repeater && ! $is_default && ! empty( $control_default_value ) ) {
+					$clear_repeater_settings = [];
+
+					// Using the control defaults, clean unused keys from $setting_value ( repeater values ).
+					foreach ( $control_default_value as $key => $control_repeater_item_default ) {
+						$clear_repeater_settings[ $key ] = array_intersect_key(
+							$setting_value[ $key ],
+							array_flip( /* allowed keys*/ array_keys( $control_repeater_item_default ) )
+						);
+					}
+
+					$is_default = $clear_repeater_settings === $control_default_value;
+				}
+
+				if ( ! $is_default ) {
+					if ( $is_repeater ) {
+						$usage[ $setting_name ] = count( $setting_value );
+					} else {
+						$usage[ $setting_name ] = 1;
+					}
+				}
+			}
+		}
```

Comment:
Suggestion:

Maybe use a collection to sperate each phase of the code:
1. filter if the settings are not in controls
2. filter if is_default
3. then map what left and add the number of usages

Another option is to set a comment before each phase to help the developer read and understand the code.

Reply:
Handled. with function name instead of comment.